operator/controllers/http/httpscaledobject_controller.go

```diff
@@ -171,3 +179,22 @@ func (r *HTTPScaledObjectReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&httpv1alpha1.HTTPScaledObject{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		Complete(r)
 }
+
+// sanitize hosts by converting the host definition to hosts are erroring
+// when both fields are set
+func sanitizeHosts(
```

Comment:
Given both fields are now optional, we should check if both are nil and raise an error when that happens. Make the Host field a `*string` so we can check when that happens.

Reply:
@t0rr3sp3dr0: I've converted the `Host` field to a `*string`, updated the `sanitizeHosts` function and added a unit test for the "both are nil" case.

Let me know if it's ok :)

Reply:
That’s perfect. Thanks!